backend/src/test/java/com/github/wekaito/backend/ProfileServiceTest.java

```diff
@@ -71,11 +72,29 @@ void testGetDecks() {
     void testUpdateDeck() {
         profileService.updateDeck(exampleDeck.id(), exampleDeckWithoutId);
         verify(deckRepo).save(exampleDeck);
+
+        when(deckRepo.existsById(exampleDeck.id())).thenReturn(false);
+        String id = exampleDeck.id();
+        assertThrows(IllegalArgumentException.class, () -> profileService.updateDeck(id, exampleDeckWithoutId));
     }
 
     @Test
     void testDeleteDeck() {
         profileService.deleteDeck(exampleDeck.id());
         verify(deckRepo).deleteById(exampleDeck.id());
+
+        when(deckRepo.existsById(exampleDeck.id())).thenReturn(false);
+        String id = exampleDeck.id();
+        assertThrows(IllegalArgumentException.class, () -> profileService.deleteDeck(id));
     }
-}
\ No newline at end of file
+
+    @Test
+    void testGetDeckById() {
+        profileService.getDeckById(exampleDeck.id());
+        verify(deckRepo).findById(exampleDeck.id());
+
+        when(deckRepo.findById(exampleDeck.id())).thenReturn(Optional.empty());
+        assertNull(profileService.getDeckById(exampleDeck.id()));
+    }
+
+}
```

Comment:
you should probably add an empty line here (as well as in the other classes and components)...maybe you did it and github deleted it or doesn't understood that you added new code? there was a discussion about it in the q&a channel...

Reply:
I did, but the depiction is confusing. 😅